apps/server/src/modules/user/legacy/controller/api-test/admin-api-students.api.spec.ts

```diff
@@ -0,0 +1,356 @@
+					$skip: 0,
+					$limit: 5,
+					$sort: { classes: 1 },
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return students', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.set('Accept', 'application/json')
+					.expect(200);
+
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(2);
+				expect(data.length).toBe(2);
+			});
+		});
+
+		describe('when sorting by consentStatus', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 0,
+					$limit: 5,
+					$sort: { consentStatus: 1 },
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return students', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.set('Accept', 'application/json')
+					.expect(200);
+
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(2);
+				expect(data.length).toBe(2);
+			});
+		});
+
+		describe('when search params are too tight', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 0,
+					$limit: 5,
+					$sort: { firstName: 1 },
+					classes: ['1A', '2A'],
+					consentStatus: { $in: ['ok', 'parentsAgreed', 'missing'] },
+					createdAt: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+					lastLoginSystemChange: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+					outdatedSince: {
+						$gt: new Date('2024-02-08T23:00:00Z'),
+						$gte: new Date('2024-02-08T23:00:00Z'),
+						$lt: new Date('2024-02-08T23:00:00Z'),
+						$lte: new Date('2024-02-08T23:00:00Z'),
+					},
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return empty list', async () => {
+				const { query } = setup();
+				const response = await request(app.getHttpServer()) //
+					.get(`${basePath}`)
+					.query(query)
+					.send()
+					.expect(200);
+				const { data, total } = response.body as UserListResponse;
+
+				expect(total).toBe(0);
+				expect(data.length).toBe(0);
+			});
+		});
+
+		describe('when skip params are too big', () => {
+			const setup = () => {
+				currentUser = mapUserToCurrentUser(adminUser, adminAccount);
+				const query: UsersSearchQueryParams = {
+					$skip: 500,
+					$limit: 5,
+					$sort: { firstName: 1 },
+				};
+
+				return {
+					query,
+				};
+			};
+
+			it('should return 200', async () => {
```

Comment:
This expectation seems too generic to me. I would expect something like 'should return empty list' or whatever happens in that case.